Utilities/Thread.cpp

```diff
@@ -2350,18 +2349,12 @@ bool thread_base::join(bool dtor) const
 			break;
 		}
 
-		if (i > 20 && Emu.IsStopped())
+		if (i && (i % 20) == 0 && timeout != atomic_wait_timeout::inf)
 		{
-			atomic_wait_engine::raw_notify(0, get_native_id());
```

Comment:
I took the liberaty to remove this hack as I think all thread code paths are CELL_OK now.